langchain_tools_demo/agent.py

```diff
@@ -70,12 +70,30 @@ def get_id_token(url: str) -> str:
         )
 
 
-def get_header() -> Optional[dict]:
+def convert_date(date_string: str) -> str:
+    """Convert date into appropriate date string"""
+    if date_string == "tomorrow":
+        converted = date.today() + timedelta(1)
+    elif date_string == "yesterday":
+        converted = date.today() - timedelta(1)
+    elif date_string != "null" and date_string != "today" and date_string is not None:
+        converted = dparser.parse(date_string, fuzzy=True).date()
+    else:
+        converted = date.today()
+
+    return converted.strftime("%Y-%m-%d")
+
+
+def get_header(user_id_token: Optional[str]) -> Optional[dict]:
```

Comment:
Why do we need get_header? Can we just set the headers when the client session is initialized?

Reply:
combined the get_headers() function with create_client_session()